OCI - SSR doc (#5978)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/concepts/self-service-registration/index.md

Self-service registration allows users to create an account and sign in to an app on their own.

A combination of policies that you can set in the Admin Console or with the Policies API lets you choose which data to collect during registration, the authenticators available for enrollment, and the authentication requirements for enrolling those authenticators. You can also use an inline hook and custom code for further customization.

For more information about SSR, see:

* [Self-service registration overview](https://help.okta.com/okta_help.htm?type=oie&id=ext-about-ssr)
* [Add a sign-up form to your web app](/docs/journeys/OCI-web-sign-up/main/)

> **Note:** If you plan to use an Okta-hosted Sign-In Widget for your sign-up form, note that the SSR flow follows the standard OpenID Connect (OIDC) protocol. Users are first redirected to an Okta-hosted page where they complete the registration form. After they register, Okta redirects them back to your app with an authorization code, which your app exchanges for tokens to sign the user in.

### Default SSR state

Self-service registration is disabled by default. There are two main policies that control SSR.

> * **User profile policies:** This policy type controls the attributes that are collected when users register and has a setting to enable self-service registration. The default user profile policy doesn't allow self-service registration.

Reviewer: ```suggestion
* **User profile policies:** This policy type controls the attributes that are collected when users register and has a setting to enable SSR. The default user profile policy doesn't allow SSR
```

* The [user profile policy](/docs/concepts/policies/#user-profile-policies) prompts users for their first name, family name, and email address. Those are the basic [Universal Directory (UD)](/docs/concepts/universal-directory/) attributes required to create a user profile and activate an Okta account.
* Users must verify their email address before the account is activated.
* Progressive enrollment is enabled. If you add other UD attributes as optional, then users are prompted to provide that information when they sign in again, after their initial registration.
* There’s one default identifier that allows users to sign in with their username. You can add other identifiers. See [Multiple identifiers](https://help.okta.com/okta_help.htm?type=oie&id=ext-multiple-ids).
* The registration form has default text and labels. You can customize the text and labels to match your brand and use case.
* The user profile policy isn't assigned to any apps. You must assign the policy to an app for users to see the registration form when they try to sign in to that app.

#### Default SSR flow diagram

> This diagram shows the default SSR flow when you enable self-service registration with a user profile policy.

Reviewer: ```suggestion
This diagram shows the default SSR flow when you enable SSR with a user profile policy.
```

> **Note:** The Okta account management policy also controls how users unenroll authenticators, recover their passwords, and unlock their accounts.

#### SSR flow diagram with policies

This diagram shows how the user profile policy, authenticator enrollment policy, and Okta account management policy work together to control the SSR flow.

* The user profile policy adds the "Business Department" field to the registration form.
* The authenticator enrollment policy makes the passkeys authenticator available during registration. You can make other authenticators available for enrollment by adding them to the relevant authenticator enrollment policy.
* The Okta account management policy requires users to verify their identity before they enroll a passkey.

<div class="three-quarter border">

![A diagram that shows an SSR flow with the user profile policy, authenticator enrollment policy, and Okta account management policy added to it.](/img/ssr-flow-policies.png)

</div>

> **Tip:** You can configure your policies in different ways to reduce the number of steps that users have to go through during registration.
>

> > For example, you can add the password field to the registration form with the user profile policy and then not require users to verify their email before they enroll the password authenticator. This allows users to register and enroll an authenticator in fewer steps.

Reviewer: ```suggestion
> For example, add the password field to the registration form with the user profile policy and then not require users to verify their email before they enroll the password authenticator. This allows users to register and enroll an authenticator in fewer steps.
```

See:

* [Style the Sign-In Widget (third generation)](/docs/guides/custom-widget-gen3/main/)
* [Unlock the Secrets of a Custom Sign-In Page with Tailwind and JavaScript](https://developer.okta.com/blog/2025/11/24/okta-custom-sign-in-page)
* [Stretch Your Imagination and Build a Delightful Sign-In Experience](https://developer.okta.com/blog/2025/11/12/custom-signin)
* [Multibrand architecture](/docs/concepts/multibrand-architecture)
* [Add design tokens to your code](/docs/guides/custom-widget-migration-gen3/main/#add-design-tokens-to-your-code)

#### How to insert custom code into the Okta-hosted widget

You must have a [custom domain](/docs/guides/custom-url-domain/main/#about-okta-domain-customization) configured to customize the Okta-hosted widget. After you configure a custom domain, you're able to insert your own scripts and styles into the registration page. And Okta continues to manage the core widget logic and security updates.

> [Use the code editor](/docs/guides/custom-widget/main/#use-the-code-editor) in the Admin Console to [add your own scripts](/docs/guides/custom-widget/main/#add-your-own-scripts) and styles to the page. You insert your custom logic by modifying the configuration before the widget renders. When using custom code, ensure your custom CSS and third-party scripts are loaded in the `<head>` of the code editor before the widget runs.

Author: Added: 
> ...ensure **that** your...
